### PR TITLE
New Resource: ServiceDiscovery PrivateDNS Namespace

### PR DESCRIPTION
**Pull request by atsushi-ishibashi**
```
TF_ACC=1 go test ./aws -v -run=TestAccAwsServiceDiscoveryPrivateDnsNamespace_basic -timeout 120m
=== RUN   TestAccAwsServiceDiscoveryPrivateDnsNamespace_basic
--- PASS: TestAccAwsServiceDiscoveryPrivateDnsNamespace_basic (69.64s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	69.713s
```
After #2569 merged, I'll modify commit logs.

### CODE REVIEW

File: aws/resource_aws_service_discovery_private_dns_namespace.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	input := &servicediscovery.DeleteNamespaceInput{
+		Id: aws.String(d.Id()),
+	}
+
+	resp, err := conn.DeleteNamespace(input)
+	if err != nil {
+		if isAWSErr(err, servicediscovery.ErrCodeNamespaceNotFound, "") {
+			d.SetId("")
+			return nil
+		}
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_private_dns_namespace.go]: This check shouldn't be necessary as we perform refresh prior to destroy and `Read()` would wipe the resource from state if it didn't exist.

File: aws/resource_aws_service_discovery_private_dns_namespace.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{servicediscovery.OperationStatusSubmitted, servicediscovery.OperationStatusPending},
+		Target:     []string{servicediscovery.OperationStatusSuccess},
+		Refresh:    servicediscoveryOperationRefreshStatusFunc(conn, *resp.OperationId),
+		Timeout:    5 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_private_dns_namespace.go]: Is there any reason for the `Delay` and `MinTimeout` here?

File: aws/resource_aws_service_discovery_private_dns_namespace.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_private_dns_namespace.go]: Is there any reason for the `Delay` and `MinTimeout` here?

File: aws/resource_aws_service_discovery_private_dns_namespace_test.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+func TestAccAwsServiceDiscoveryPrivateDnsNamespace_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsServiceDiscoveryPrivateDnsNamespaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceDiscoveryPrivateDnsNamespaceConfig(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsServiceDiscoveryPrivateDnsNamespaceExists("aws_service_discovery_private_dns_namespace.test"),
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_private_dns_namespace_test.go]: As mentioned in your other PR - can we add more checks here?

**Comment by atsushi-ishibashi** [comment]
@radeksimko Ok!
```
TF_ACC=1 go test ./aws -v -run=TestAccAwsServiceDiscoveryPrivateDnsNamespace_basic -timeout 120m
=== RUN   TestAccAwsServiceDiscoveryPrivateDnsNamespace_basic
--- PASS: TestAccAwsServiceDiscoveryPrivateDnsNamespace_basic (205.51s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	205.581s
```

**Comment by radeksimko** [comment]
Hey @atsushi-ishibashi 
I just merged #2569 - do you mind rebasing here?

Thanks.

**Comment by atsushi-ishibashi** [comment]
@radeksimko OK👍

**Pull request merged**